DOCSP-4305 Write data (#67)
mongodb/docs-c: source/write.txt

Write Data to MongoDB
=====================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference
 
.. meta::

>    :description: Learn how to use {+driver-short+} to write data to MongoDB.

Reviewer: S:
```suggestion
   :description: Learn how to use the {+driver-short+} to write data to MongoDB.
```

   /write/update
   /write/delete
   /write/bulk-write

Overview
--------

> On this page, you can see copyable code examples that show common methods you

Reviewer: S: use "functions" instead of "methods"
```suggestion
On this page, you can see copyable code examples that show common functions you
```

can use to write data to MongoDB with {+driver-short+}.

.. tip::

>   To learn more about any of the methods shown on this page, see the link

Reviewer: S:
```suggestion
  To learn more about any of the functions shown on this page, see the link
```

To use an example from this page, copy the code example into the :ref:`sample
application <c-write-sample>` or your own application. Be sure to replace all
placeholders in the code examples, such as ``<connection string URI>``, with the
relevant values for your MongoDB deployment.

.. _c-write-sample:

.. include:: /includes/usage-examples/sample-app-intro.rst

.. literalinclude:: /includes/usage-examples/sample-write-app.c
    :language: c
    :copyable:
    :linenos:
    :emphasize-lines: 20-22

Insert One 
----------

The following code shows how to insert a single document into a collection:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-insert-one
   :end-before: end-insert-one
   :language: c
   :copyable:
   :dedent:

To learn more about the ``mongoc_collection_insert_one()`` function, see the
:ref:`Insert Documents <c-write-insert>` guide.

Insert Multiple
---------------

The following code shows how to insert multiple documents into a collection:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-insert-many
   :end-before: end-insert-many
   :language: c
   :copyable:
   :dedent:

> To learn more about the ``mongoc_collection_insert_one()`` function, see the

Reviewer: S:
```suggestion
To learn more about the ``mongoc_collection_insert_many()`` function, see the
```

:ref:`Insert Documents <c-write-insert>` guide.

Update One
----------

The following code shows how to update a single document in a collection by
creating or editing a field:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-update-one
   :end-before: end-update-one
   :language: c
   :copyable:
   :dedent:

> To learn more about the ``mongoc_collection_insert_one()`` function, see the

Reviewer: S:
```suggestion
To learn more about the ``mongoc_collection_update_one()`` function, see the
```

:ref:`Update Documents <c-write-update>` guide.

Update Multiple
---------------

The following code shows how to update multiple documents in a collection by
creating or editing a field:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-update-many
   :end-before: end-update-many
   :language: c
   :copyable:
   :dedent:

> To learn more about the ``mongoc_collection_insert_many()`` function, see the

Reviewer: S:
```suggestion
To learn more about the ``mongoc_collection_update_many()`` function, see the
```

:ref:`Update Documents <c-write-update>` guide.

Delete One
----------

The following code shows how to delete a single document in a collection:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-delete-one
   :end-before: end-delete-one
   :language: c
   :copyable:
   :dedent:

To learn more about the ``mongoc_collection_delete_one()`` function, see the
:ref:`Delete Documents <c-write-delete>` guide.

Delete Multiple
---------------

The following code shows how to delete multiple documents in a collection:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-delete-many
   :end-before: end-delete-many
   :language: c
   :copyable:
   :dedent:

To learn more about the ``mongoc_collection_delete_many()`` function, see the
:ref:`Delete Documents <c-write-delete>` guide.

Bulk Write 
----------

> The following code shows how to perform multiple write operations in a single

Reviewer: S: this code example only shows one write operation (insert many); maybe add another type of write operation, and call the ``mongoc_bulk_operation_execute()`` function to run the bulk operation

Author: Made some updates, let me know if that's more aligned with what you're thinking. Thanks for this suggestion!

bulk operation:

.. literalinclude:: /includes/usage-examples/write-code-examples.c
   :start-after: start-bulk-write
   :end-before: end-bulk-write
   :language: c
   :copyable:
   :dedent:

> To learn more about the ``mongoc_collection_bulk_operation_with_opts()``

Reviewer: S: if you edit the code example to use mongoc_bulk_operation_execute(), I'd mention that function here instead
```suggestion
To learn more about the ``mongoc_bulk_operation_execute()``
```

   /write/update
   /write/delete
   /write/bulk-write

Overview
--------

On this page, you can see copyable code examples that show common functions you

> can use to write data to MongoDB with {+driver-short+}.

Reviewer: S:
```suggestion
can use to write data to MongoDB with the {+driver-short+}.
```
